src/EnergyPlus/ChillerElectricEIR.hh

```diff
@@ -187,7 +188,14 @@ namespace ChillerElectricEIR {
         Real64 CondenserFanEnergyConsumption = 0.0; // reporting: Air-cooled condenser fan energy [J]
         Real64 BasinHeaterConsumption = 0.0;        // Basin heater energy consumption (J)
         bool IPLVFlag = true;
+        int ChillerCondLoopFlowFLoopPLRIndex = 0; // Condenser loop flow rate fraction function of loop PLR
+        int CondDT = 0;                           // Temperature difference across condenser
+        int CondDTScheduleNum = 0;                // Temperature difference across condenser schedule index
+        Real64 MinCondFlowRatio = 0.2;            // Minimum condenser flow fraction
```

Comment:
This defaults to 0.2

Reply:
In the good old days when one could edit the IDD and change the default this sort thing made sense to me.  Now that the IDD is hard coded in the JSON stuff what is the difference if it is hard coded in C++ or JSON?  Just saying

RTFM?
![image](https://github.com/user-attachments/assets/75246a07-4e02-4e0e-a9c0-c01700ae1968)

Reply:
Addressed in https://github.com/NREL/EnergyPlus/pull/10779.

Reply:
Okay, IDD as documentation makes good sense.  But I am curious if it is okay to hardcode the default as initialization in C++ variable declaration or if the preferred practice is to handle the default value coming from input processor.

Reply:
Totally agree with @mjwitte here.

Also, I've made that point several times over the years, but in my opinion, every field should be either `\required-field` or `\default` _unless there's a very good reason_ (examples: no default value is appropriate, and it's a field that's only used based on some other choice field; or an optional loop inlet/outlet nodes). This is **very** helpful for anyone consuming the IDD (such as OpenStudio SDK, and IDF / EPJSON editors).

`\mininimum` and `\maximum` and the `\type` attributes should be used as much as possible on the fields too.

And `\extensible` should be used on the object itself everywhere it's appropriate.

**The IDD is a schema**  (and the fact it's translated into the epJSON schema makes that indisputable), **so it should be as good as possible**, without having to RTFM or even worse go through the C++ code

---

Note: In the OpenStudio SDK:

* We have an our own IDD for model, and we have a modified Energy+.idd for workspace (IDF), but when a new E+ version comes out, we take the official E+ idd as a basis to bring in changes. Everytime the IDD changes are lacking proper flags, we have to read the documentation AND go through the C++ code. 
     * We know that the IDD is lacking proper flags because we're more stringent with them:
          * The primary reason is that we have C++ code generators scripts that will read the IDD and implement getters/setters differently based on whether a field is required, defaulted, or nothing.
          * The second reason is that we try hard to avoid people shooting themselves in the foot
* We differentiate between a defaulted value and a user-set value (the getter returns either the defaulted or user-set value, but you have a `bool isValueDefaulted()` and a `resetValue()` method as well, and interfaces such as the OpenStudioApp will display that field differently (green if defaulted for eg)) 
* More often than not (for HVAC objects especially), we turn a `\default xxx` field in a `\required-field` and hardcode the value in the object's ctor, as this simplifies the API (saves writing and maintaining a getter and a resetter)